src/types.ts

```diff
@@ -8,9 +8,15 @@ export type DMMF = typeof Prisma.dmmf
 
 // Internal types --
 
+export type EncryptionFn = (value: string) => string
+
+export type DecryptionFn = (value: string) => string
+
 export interface Configuration {
   encryptionKey?: string
   decryptionKeys?: string[]
+  encryptionFn?: EncryptionFn
```

Comment:
Since providing a custom encryption method most likely results in having to provide the corresponding decryption method, we should enforce this using TypeScript, as such:

```ts
export interface Configuration = {
  // ...
  cipher?: {
    encrypt: (clearText: string) => string
    decrypt: (cipherText: string) => string
  }
}
```

Reply:
I understand your request and in fact it make perfect sense. I just updated the branch with the requested changes.